src/main/kotlin/baseball/GameHandler.kt

```diff
@@ -0,0 +1,96 @@
+package baseball
+
+import baseball.data.GameNumDeck
+import baseball.data.InputType
+
+class GameHandler(private val io: IOHandler) {
+
+    private lateinit var answer: GameNumDeck
+    private var strikeCount = 0
+    private var ballCount = 0
+
+    fun start() {
+        io.show(SENTENCE_FOR_START)
+        var isKeepPlay = YES
+
+        while (isKeepPlay == YES) {
+            reset()
+            playGame()
+
+            io.show(SENTENCE_FOR_AFTER_GAME)
+            isKeepPlay = io.getInput(InputType.AFTER_GAME).toInt()
+        }
+    }
+
+    private fun reset() {
+        answer = GameNumDeck.generateRandomDeck()
+    }
+
+    private fun playGame() {
+        io.show(SENTENCE_FOR_INPUT)
+
+        var inputString = io.getInput(InputType.WHILE_GAME)
+        while (checkInputIsCorrect(inputString).not()) {
+            io.show(SENTENCE_FOR_INPUT)
+            inputString = io.getInput(InputType.WHILE_GAME)
+        }
+
+        io.show(SENTENCE_FOR_CLOSING)
+    }
+
+    private fun checkInputIsCorrect(inputString: String): Boolean {
+        resetStrikeAndBallCount()
+
+        val nums = splitInputToNumber(inputString)
+        for ((index, num) in nums.withIndex()) {
+            checkIsStrikeOrBall(index, num)
+        }
```

Comment:
```suggestion
        nums.mapIndexed { index, num ->
            checkIsStrikeOrBall(index, num)
        }
```
[코드 컨벤션](https://kotlinlang.org/docs/coding-conventions.html#loops) : **루프(for)보다 고차(higher-order functions) 함수(`filter`, `map, reduce` etc.)를 사용**을 권장합니다!

Reply:
map은 배열에 있는 값을 변환해서 반환하기 때문에, 리턴 되는 값이 필요하지 않은 해당 부분에 어울리지 않다고 생각해요.

forEachIndexed는 컨벤션 주의사항에 단독으로 사용될 경우, for로 구현하라고 되어 있어 사용하지 않았습니다.

Reply:
아하 그렇군요 처음 알았네요. 말씀해주신 것처럼 위 for문은 단순한 반복의 역할만 수행하기 때문에 map을 사용하면 필요없는 메모리만 더 사용하게 되겠군요!
덕분에 map 사용의 목적을 정의할 수 있었습니다!
-> `콜렉션의 각 요소를 변환하는 작업이 필요할 때 map을 사용하자!`